Failing to issue a certificate should not prevent other certificates from running

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->
```

Requesting a certificate for commit-composer.dev and 2 more domains
--
An unexpected error occurred:
Error creating new order :: too many certificates (5) already issued for this exact set of domains in the last 168 hours: commit-composer.dev,staging.commit-composer.dev,www.commit-composer.dev, retry after 2024-01-29T15:13:04Z: see https://letsencrypt.org/docs/duplicate-certificate-limit/
Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /tmp/logs-dir/letsencrypt.log or re-run Certbot with -v for more details.
[ERROR]	2024-01-28T06:40:14.202Z	cb697a23-9685-41af-9d7c-038881e8fd21	[Errno 2] No such file or directory: '/tmp/config-dir/live/commit-composer.dev/cert.pem'
[ERROR] FileNotFoundError: [Errno 2] No such file or directory: '/tmp/config-dir/live/commit-composer.dev/cert.pem'Traceback (most recent call last):  File "/var/task/main.py", line 117, in handler    cert = provision_cert(email, lineage, domains)  File "/var/task/main.py", line 53, in provision_cert    'certificate': read_and_delete_file(path + 'cert.pem'),  File "/var/task/main.py", line 30, in read_and_delete_file    with open(path, 'r') as file:
END RequestId: cb697a23-9685-41af-9d7c-038881e8fd21
REPORT RequestId: cb697a23-9685-41af-9d7c-038881e8fd21	Duration: 3637.22 ms	Billed Duration: 3638 ms	Memory Size: 512 MB	Max Memory Used: 124 MB
```

--- a/src/main.py
+++ b/src/main.py
@@ -106,16 +106,22 @@ def upload_cert_to_acm(cert, domains):
 
     return None
 
+def process_lineage(lineage, domains, email):
+    try:
+        logger.info('Processing: ({}) {}'.format(lineage, domains))
+        if should_provision(domains):
+            cert = provision_cert(email, lineage, domains)
+            upload_cert_to_acm(cert, domains)
+    except Exception as e:
+        logger.error(e)
+
 def handler(event, context):
     try:
         dlist = DomainList(os.environ['DOMAIN_LIST'])
         email = os.environ['DOMAIN_EMAIL']
         logger.info('Processing domain list: {}'.format(dlist.original))
         for lineage, domains in dlist.lineage.items():
-            logger.info('Processing: ({}) {}'.format(lineage, domains))
-            if should_provision(domains):
-                cert = provision_cert(email, lineage, domains)
-                upload_cert_to_acm(cert, domains)
+            process_lineage(lineage, domains, email)
     except Exception as e:
         logger.error(e)
         raise

